### PR TITLE
Graph refreshing functionality

### LINKED ISSUE #11: Implement safe state saving and update functionality
Use flask caching to safely save state on server side. Use refresh button to read data from disk and update the graph status accordingly.

### PR DESCRIPTION
**Pull request by riverchen99**
This PR implements the safe server side state storage approach described in the implementation document. The major interesting changes are in app.py. Other changes are related to a modification I had to make to the protobuf import mechanism to support caching.

closes #11

### CODE REVIEW

File: ujt/app.py [hunk 1]
```diff
@@ -17,31 +17,47 @@
...
+cache.init_app(
+    app.server,
+    config={
+        "CACHE_TYPE": "filesystem",
+        "CACHE_DIR": "cache_dir"
+    },
+)
+
 
+@cache.memoize()
```
Comment (riverchen99) [inline ujt/app.py:45]: Here is the interesting part -- memoize the call to read_local_data() so we don't actually read from disk (in the future, server) unless the cache is invalidated.

File: ujt/app.py [hunk 2]
```diff
@@ -75,33 +91,37 @@ def read_local_data() -> Tuple[Dict[str, Node], Dict[str, Client]]:
...
+    """ Gets a dictionary mapping Client names to Client messages.
+
+    Currently reads local data, but should be eventually modified to use remote data.
 
     Returns:
-        A list of dictionary objects, each containing information regarding a single node (Service or Client) or edge (Dependency).
+        A dictionary mapping Client names to Client messages
     """
-    raise NotImplementedError
+    return read_local_data()[1]
```
Comment (riverchen99) [inline ujt/app.py:124]: This call is memoized.

File: ujt/app.py [hunk 3]
```diff
@@ -116,21 +136,21 @@ def generate_graph_elements(node_name_message_map: Dict[str, Node],
...
-               config={
-                   "CACHE_TYPE": "filesystem",
-                   "CACHE_DIR": "cache_dir"
-               })
+@app.callback(
+    Output("cytoscape-graph", "elements"),
+    [Input("refresh-button", "n_clicks_timestamp")],
+)
+def refresh_graph(n_clicks_timestamp):
+    cache.clear()
```
Comment (riverchen99) [inline ujt/app.py:148]: Force the cache to be cleared on button click. In the future, I will attach this callback to an [interval](https://dash.plotly.com/dash-core-components/interval) to refresh the graph automatically.

**Review by pwat2 — APPROVED** [review]

**Comment by riverchen99** [comment]
In a later branch, I refactored these methods (read_local_data, get_map) into a separate state.py file so that all parts of the code can have a unified way to access the current state of the application.

**Review by KenArthur — APPROVED** [review]

**Pull request merged**